src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -305,6 +305,35 @@ public override async Task<bool> CheckHealthAsync(CancellationToken cancellation
             }
         }
 
+        public override async Task<bool> CheckOutboundHealthAsync(CancellationToken cancellationToken = default)
+        {
+            var path = "/v1.0/healthz/outbound";
+            var request = new HttpRequestMessage(HttpMethod.Get, new Uri(this.httpEndpoint, path));
+            try
+            {
+                var response = await this.httpClient.SendAsync(request, cancellationToken);
+                return response.IsSuccessStatusCode;
+            }
+            catch (HttpRequestException)
+            {
+                return false;
+            }
+        }
+
+        public override async Task<bool> WaitForSidecarAsync(CancellationToken cancellationToken = default)
+        {
+            while (!cancellationToken.IsCancellationRequested)
+            { 
+                var response = await CheckOutboundHealthAsync(cancellationToken);
```

Comment:
This code has an issue because **cancellation** can be signaled in two different ways:

1. We detect cancellation in line 325 and return false
2. Cancellation occurs while the request is trying to connect and results in an exception

I think the best fix is to remove the boolean value on wait and just always throw an exception. There's a method on cancellation token that will throw if it's canceled, so we can change line 325 to do that

Reply:
Done, I was trying to be polite and eat the exception but after chatting it's clear that the exception is actually the expected case.